core/src/io/anuke/mindustry/world/blocks/power/PowerNode.java

```diff
@@ -299,4 +307,27 @@ protected void drawLaser(Tile tile, Tile target){
         Draw.color();
     }
 
+    public static boolean insulated(Tile tile, Tile other){
+        return insulated(tile.x, tile.y, other.x, other.y);
+    }
+
+    public static boolean insulated(int x, int y, int x2, int y2){
+        final Boolean[] bool = {false};
+        insulators(x, y, x2, y2, cause -> {
+            bool[0] = true;
+        });
+        return bool[0];
+    }
+
+    public static void insulators(int x, int y, int x2, int y2, Consumer<Tile> iterator){
+        world.raycastEach(x, y, x2, y2, (wx, wy) -> {
+
+            Tile tile = world.ltile(wx, wy);
+            if(tile != null && tile.block() != null && tile.block().insulated){
+                iterator.accept(tile);
```

Comment:
Once you hit an insulator, it may be a good idea to `return true` so the raycasting stops. `tile.block()` can't be null, so the condition here is redundant.

Reply:
But then you cannot highlight all the walls that cause the insulation 😗